README.md

```diff
@@ -60,6 +60,26 @@ The rest of the module is usually very minimalistic:
 
 Please note: we currently have modules that use `apypie` and `nailgun` as the backend libraries to talk to the API, but we would prefer not to add any new modules using `nailgun` and focus on migrating everything to `apypie`.
 
+### Specification of the `entity_spec`
+
+The `entity_spec` can be seen as an extended version of Ansible's `argument_spec`. It understands more parameters (e.g. `flat_name`) and supports more `type`s than the original version. An `argument_spec` will be generated from a `entity_spec`, any parameters not directly known or consumed by `entity_spec` will be passed directly to the `argument_spec`.
+
+In addition to Ansible's `argument_spec`, `entity_spec` understands the following types:
+
+* `type='entity'` The referenced value is another Foreman entity.
+This is usually combined with `flat_name=<entity>_id`.
+* `type='entity_list'` The referenced value is a list of Foreman entities.
+This is usually combined with `flat_name=<entity>_ids`.
+* `type='nested_list'` The referenced value is a list of Foreman entities that are not included in the main API call.
+The module must handle the entities separately.
+See domain parameters for an example.
```

Comment:
Perhaps domain could be a link? Otherwise make it clearer that this is the domain entity?